harvester_e2e_tests/scenarios/test_create_vm.py

```diff
@@ -219,6 +220,51 @@ def test_create_update_vm_machine_type(request, admin_session, image, keypair,
                 utils.delete_vm(request, admin_session,
                                 harvester_api_endpoints, vm_json)
 
+
+def test_create_update_vm_user_data(request, admin_session, image, single_vm,
+                                    harvester_api_endpoints,
+                                    user_data_with_guest_agent, network_data):
+    # make sure the VM instance is successfully created
+    vm_instance_json = utils.lookup_vm_instance(
+        admin_session, harvester_api_endpoints, single_vm)
+    vm_name = single_vm['metadata']['name']
+    previous_uid = vm_instance_json['metadata']['uid']
+    vol_data = single_vm['spec']['template']['spec']['volumes']
+    assert 'cloudInitNoCloud' not in vol_data
+    userdata = user_data_with_guest_agent.replace('\\n', '\n')
+    netdata = network_data.replace('\\n', '\n')
```

Comment:
I don't we need to do this as the fixture has already returning the correct format. See
https://github.com/harvester/tests/blob/main/harvester_e2e_tests/fixtures/vm.py#L44